### PR TITLE
Extend the subset_vds.py to be able to subset on a sample list

### PR DESCRIPTION
**Pull request by katiedelange**
The main change here is adding an optional argument to filter the VDS using a space-separated list of sample IDs.

A few additional changes:
- Allow the user to adjust the suffix used for the output files (default:  "_subset")
- Genotype information should be stored in the base buckets, rather than analysis
- Fixed an arguments bug with filter_variants

### CODE REVIEW

File: subset_vds.py [hunk 1]
```diff
@@ -335,14 +340,14 @@ def main(
         else:
             subset_vds = subset_by_variants(input_variants, keep_variants, input_vds)
 
-    if n_samples:
-        subset_sample_list = get_subset_sample_list(input_vds, n_samples)
+    if n_samples or sample_list:
+        subset_sample_list = sample_list or get_subset_sample_list(input_vds, n_samples)
```
Comment (Alexander-Stuckey) [inline subset_vds.py:344]: What happens here if someone provides both a sample list and `n_samples`? I suggest that we add a check earlier to only allow one.

**Review by Alexander-Stuckey — CHANGES_REQUESTED** [review]
Suggest a check to allow only one of sample list or `n_samples`. The rest looks good to me (and thanks for fixing the issue with `filter_variants`)

File: subset_vds.py [hunk 2]
```diff
@@ -385,16 +403,25 @@ def main(
...
+        and not args.sample_list
+        and not args.intervals
+        and not args.variant_table
+    ):
         raise AttributeError(
-            "If neither a number of samples or intervals are provided, this script will not do any subsetting!"
+            "If no filtering fields (n_samples, sample_list, intervals and/or variant_table) are provided, this script will not do any subsetting!"
         )
+    if args.n_samples and args.sample_list:
+        raise AttributeError("You can only provide one of n_samples or sample_list.")
```
Comment (katiedelange) [inline subset_vds.py:416]: I think this line was already catching that scenario?

**Review by Alexander-Stuckey — APPROVED** [review]
Looks good to me

**Pull request merged**